observatory-platform/observatory/platform/utils/gc_utils.py

```diff
@@ -21,28 +21,164 @@
 import multiprocessing
 import os
 import re
+import sys
 import time
 from concurrent.futures import ProcessPoolExecutor, as_completed
+from copy import deepcopy
 from enum import Enum
 from multiprocessing import BoundedSemaphore, cpu_count
-from typing import List, Tuple, Union
+from typing import Any, List, Optional, Tuple, Union
 
 import pendulum
+from airflow.models import Variable
 from google.api_core.exceptions import BadRequest, Conflict
 from google.cloud import bigquery, storage
 from google.cloud.bigquery import LoadJob, LoadJobConfig, QueryJob, SourceFormat
+from google.cloud.bigquery.job import QueryJobConfig
 from google.cloud.exceptions import Conflict, NotFound
 from google.cloud.storage import Blob
 from googleapiclient import discovery as gcp_api
-from requests.exceptions import ChunkedEncodingError
-
+from observatory.api.client.exceptions import NotFoundException
+from observatory.api.client.model.big_query_bytes_processed import (
+    BigQueryBytesProcessed,
+)
+from observatory.platform.utils.airflow_utils import AirflowVars, check_variables
+from observatory.platform.utils.api import make_observatory_api
 from observatory.platform.utils.config_utils import utils_templates_path
 from observatory.platform.utils.file_utils import crc32c_base64_hash
-from observatory.platform.utils.jinja2_utils import make_sql_jinja2_filename, render_template
+from observatory.platform.utils.jinja2_utils import (
+    make_sql_jinja2_filename,
+    render_template,
+)
+from requests.exceptions import ChunkedEncodingError
 
 # The chunk size to use when uploading / downloading a blob in multiple parts, must be a multiple of 256 KB.
 DEFAULT_CHUNK_SIZE = 256 * 1024 * 4
 
+# BigQuery daily query byte limit.
+BIGQUERY_QUERY_DAILY_BYTE_LIMIT = 1024 * 1024 * 1024 * 1024 * 5  # 5 TiB
+
+# Disable for now.  Tests across the different *-workflows repos need changing.
+
+
+def bq_query_daily_limit_enabled():
```

Comment:
Yeah for easy mocking in tests.  Once confident it's working properly in deployment for a while and you're sure you always want it on, this can just be removed entirely.

Reply:
Could you add this as a comment to the function?